### PR TITLE
Add DeleteFrom support for Spark 3.0

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

File: spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java [hunk 1]
```diff
@@ -119,4 +124,16 @@ public WriteBuilder newWriteBuilder(CaseInsensitiveStringMap options) {
     return new SparkWriteBuilder(sparkSession(), icebergTable, options);
   }
 
+  @Override
+  public void deleteWhere(Filter[] filters) {
+    icebergTable.newDelete()
+        .set("spark.app.id", sparkSession().sparkContext().applicationId())
+        .deleteFromRowFilter(SparkFilters.convert(filters))
```
Comment (jerryshao) [inline spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java]: IIUC, current `deleteFromRowFilter` should match all the contained rows, whereas Spark's `deleteWhere` seems like a row-level delete semantic, so seems like it cannot fully support Spark's DELETE clause, am I right?

Comment (rdblue) [reply spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java]: Implementations may reject a delete if it requires "significant effort", from the Spark docs:

> Implementations may reject a delete operation if the delete isn't possible without significant effort.

That means we can throw `IllegalArgumentException` here if the commit fails. Looks like I missed that commit will throw a `ValidationException`, not `IllegalArgumentException`, so I'll add a fix to convert it to the expected exception type.

Comment (jerryshao) [reply spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java]: I see, but it also means that Iceberg cannot fully support Spark `deleteWhere` currently, the behavior may be unexpected. Maybe we should add some comments or specific exception logs to notify users.

Comment (rdblue) [reply spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java]: The error message is fairly clear. It says that there was a file that may contain both records that should be deleted and records that should not be.

Comment (rdblue) [reply spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java]: The error message is fairly clear. It says that there was a file that may contain both records that should be deleted and records that should not be.

**Comment by chenjunjiedada** [comment]
LGTM +1

**Comment by rdblue** [comment]
Thanks for reviewing, @chenjunjiedada and @jerryshao!

**Pull request merged**